aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -146,54 +209,180 @@ Transient lifetime services are created each time they're requested. This lifeti
+
+> [!WARNING]
+> It's dangerous to resolve a scoped service from a singleton. It may cause the service to have incorrect state when processing subsequent requests.
+
+### Constructor injection behavior
+
+Services can be resolved by two mechanisms:
+
+* `IServiceProvider`
+* [ActivatorUtilities](/dotnet/api/microsoft.extensions.dependencyinjection.activatorutilities)
+
+When services are resolved by `IServiceProvider` or `ActivatorUtilities`, constructor injection requires a *public* constructor. Otherwise, the app throws an [InvalidOperationException](/dotnet/api/system.invalidoperationexception):
+
+> A suitable constructor for type '&lt;TYPE&gt;' couldn't be located. Ensure the type is concrete and services are registered for all parameters of a public constructor.
+
+When services are resolved by `ActivatorUtilities`, constructor injection requires that only one applicable constructor exist. Constructor overloads are supported, but only one overload can exist whose arguments can all be fulfilled by dependency injection. If more than one exists, the app throws an `InvalidOperationException`:
+
+> Multiple constructors accepting all given argument types have been found in type '&lt;TYPE&gt;'. There should only be one applicable constructor.
+
+Constructors can accept arguments that aren't provided by dependency injection, but these must support default values.
+
+The following code throws an `InvalidOperationException` because the runtime is unable to resolve the `item` string:
+
+```csharp
+public HomeController(IMyDependency myDependency, string item)
+{
+    _myDependency = myDependency;
+    _item = item;
+}
+```
+
+The following code runs without an exception:
+
+```csharp
+public HomeController(IMyDependency myDependency, string message = "Jelly Baby")
+{
+    _myDependency = myDependency;
+    _item = item;
+}
+```
+
+## Entity Framework contexts
+
+Entity Framework contexts should be added to the service container using the scoped lifetime. This is handled automatically with a call to the [AddDbContext](/dotnet/api/microsoft.extensions.dependencyinjection.entityframeworkservicecollectionextensions.adddbcontext) method when registering the database context. Services that use the database context should also use the scoped lifetime.
+
+## Lifetime and registration options
+
+To demonstrate the difference between the lifetime and registration options, consider the following interfaces that represent tasks as an operation with a unique identifier, `OperationId`. Depending on how the lifetime of a service is configured, the container provides either the same or a different instance of the service to the requesting class.
```

Comment:
> Depending on how the lifetime of a service is configured, the container provides either the same or a different instance of the service to the requesting class.

Can you say
The  `OperationId` depends on the service lifetime.  X, Y, Z return a new instance of  `OperationId`. M, X returns ...

Reply:
Idk about this ... the statement is accurate and the extensions are both obvious and previously covered in the *Service lifetimes* section just above this section.

I would like to clarify it to ...

> Depending on how the lifetime of an operations service is configured for the following interfaces, the container provides either the same or a different instance of the service when requested by a class:

... which makes it a hair clearer that the following interfaces will be used to create services shortly and that service lifetime registrations will dictate receiving a new or existing instance. The naming of the interfaces is obvious on which one goes with which lifetime registration.